src/Symfony/Component/Serializer/Exception/MissingConstructorArgumentsException.php

```diff
@@ -12,37 +12,66 @@
 namespace Symfony\Component\Serializer\Exception;
 
 /**
- * @deprecated since Symfony 6.3, use {@see MissingConstructorArgumentException} instead
- *
  * @author Maxime VEBER <maxime.veber@nekland.fr>
  */
 class MissingConstructorArgumentsException extends RuntimeException
 {
     /**
-     * @var string[]
+     * @var string|null
+     */
+    private $class;
+
+    /**
+     * @var MissingConstructorArgumentException[]
      */
-    private $missingArguments;
+    private $missingConstructorArgumentExceptions = [];
 
-    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [])
+    /**
+     * @param MissingConstructorArgumentException[] $missingArguments
+     * @param ?class-string $class
+     */
+    public function __construct(string $message, int $code = 0, \Throwable $previous = null, array $missingArguments = [], string $class = null)
     {
-        if (!$this instanceof MissingConstructorArgumentException) {
-            trigger_deprecation('symfony/serializer', '6.3', 'The "%s" class is deprecated, use "%s" instead.', __CLASS__, MissingConstructorArgumentException::class);
+        $this->missingConstructorArgumentExceptions = $missingArguments;
+        $classes = array_unique(array_map(fn (MissingConstructorArgumentException $missingConstructorArgumentException) => $missingConstructorArgumentException->getClass(), $missingArguments));
```

Comment:
This is still a breaking change, any existing code that passes an array of strings will fail.

Reply:
I've reworked that and added test to be sure that both options are allowed but strings shall trigger a deprecation now.